astropy/utils/introspection.py

```diff
@@ -136,30 +136,13 @@ def minversion(module, version, inclusive=True, version_path='__version__'):
     else:
         raise ValueError('module argument must be an actual imported '
                          'module, or the import name of the module; '
-                         'got {!r}'.format(module))
-
-    if '.' not in version_path:
-        have_version = getattr(module, version_path)
-    else:
-        have_version = resolve_name(module.__name__, version_path)
-
-    # LooseVersion raises a TypeError when strings like dev, rc1 are part
-    # of the version number. Match the dotted numbers only. Regex taken
-    # from PEP440, https://www.python.org/dev/peps/pep-0440/, Appendix B
-    expr = '^([1-9]\\d*!)?(0|[1-9]\\d*)(\\.(0|[1-9]\\d*))*'
-    m = re.match(expr, version)
-    if m:
-        version = m.group(0)
-
-    # have_version can have the same issue as version, so also regex it
-    m = re.match(expr, have_version)
-    if m:
-        have_version = m.group(0)
+                         f'got {repr(module)}')
 
+    module_version = metadata.version(module_name)
     if inclusive:
-        return LooseVersion(have_version) >= LooseVersion(version)
+        return module_version >= version
```

Comment:
`importlib.metadata.version` returns a string, not a version object like `LooseVersion` or [`packaging.version.Version`](https://packaging.pypa.io/en/latest/version.html). Strings don't compare well as versions, e.g. `'3.10' < '3.9'`.